packages/headless/src/database/mongodb/order/index.ts

```diff
@@ -1,32 +1,78 @@
-import { 
-  ChangeStatusEntity, 
-  GetAllOrderQueryEntity, 
-  OrderEntity, 
-  OrderIncompleteStatEntity, 
-  OrderSortQuery, 
-  OrderStatEntity, 
-  OrderStatusEnum, 
-  ShippingStatusEnum, 
-  StatusTypeDto 
+import {
+  CartItem, 
+  ReOrderQuery,
+  ChangeStatusEntity,
+  GetAllOrderQueryEntity,
+  OrderEntity,
+  OrderIncompleteStatEntity,
+  OrderSortQuery,
+  OrderStatEntity,
+  OrderStatusEnum,
+  ProductOrder,
+  ShippingStatusEnum,
+  StatusTypeDto,
 } from 'src/entity/order';
 import { IOrderDatabase } from 'src/modules/order/repositories/order.db.interface';
 import { ProductModel } from '../product/product.model';
 import { OrderModel } from './order.model';
-import { CreateOrderRequest, CreateProductOrderDetails, IProductOrderData } from 'models';
+import {
+  CreateOrderRequest,
+  CreateProductOrderDetails
+} from 'models';
+import { CartModel } from '../cart/cart.model';
 
 export class OrderDatabase implements IOrderDatabase {
+  async addToCart(userId: string, items: CartItem[]) : Promise<any> {
```

Comment:
change function name. this is not simple 'add to cart'. here you are populating cart from past order items. so you may name the function like 'populateItemsInCart'